disperser/dataapi/metrics.go

```diff
@@ -236,3 +245,157 @@ func (collector *DynamoDBCollector) Collect(ch chan<- prometheus.Metric) {
 		disperser.Processing.String(),
 	)
 }
+
+// BlobStatusMetrics holds the metrics for a specific blob status
+type BlobStatusMetrics struct {
+	gauge        prometheus.Gauge
+	currentValue float64
+}
+
+// DynamoDBCollectorV2 collects metrics about blob metadata from DynamoDB.
+type DynamoDBCollectorV2 struct {
+	blobMetadataStore *blobstorev2.BlobMetadataStore
+	statusMetrics     map[commonv2.BlobStatus]*BlobStatusMetrics
+	logger            logging.Logger
+	ctx               context.Context
+	cancel            context.CancelFunc
+}
+
+func NewDynamoDBCollectorV2(blobMetadataStore *blobstorev2.BlobMetadataStore, logger logging.Logger) *DynamoDBCollectorV2 {
+	ctx, cancel := context.WithCancel(context.Background())
+	collector := &DynamoDBCollectorV2{
+		blobMetadataStore: blobMetadataStore,
+		statusMetrics:     make(map[commonv2.BlobStatus]*BlobStatusMetrics),
+		logger:            logger,
+		ctx:               ctx,
+		cancel:            cancel,
+	}
+
+	// Create a gauge for each possible status (that is not terminal)
+	for _, status := range []commonv2.BlobStatus{
+		commonv2.Queued,
+		commonv2.Encoded,
+		commonv2.GatheringSignatures,
+	} {
+		gauge := prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Name: "dynamodb_blob_metadata_v2_status_count",
```

Comment:
Drop "dynamodb"? If we put this metric at dataapi, make it agnostic to database choice for blob status tracking

Reply:
So instead of DynamoDB collector we call it BlobMetadataStoreCollector etc.? I will rename the metric to `eigenda_blob_metadata_store_status_count`.

Reply:
yea sg